### PR TITLE
Add an "ostree config unset" operation

### PR DESCRIPTION
**Pull request by mwleeds**

### CODE REVIEW

File: man/ostree-config.xml [hunk 1]
```diff
@@ -62,20 +62,42 @@ Boston, MA 02111-1307, USA.
             <cmdsynopsis>
                 <command>ostree config set</command> <arg choice="req"> --group=GROUPNAME</arg> <arg choice="req"> KEYNAME</arg> <arg choice="req">VALUE</arg>
             </cmdsynopsis>
+            <cmdsynopsis>
+                <command>ostree config unset</command> <arg choice="req">SECTIONNAME.KEYNAME</arg>
+            </cmdsynopsis>
+            <cmdsynopsis>
+                <command>ostree config unset</command> <arg choice="req"> --group=GROUPNAME</arg> <arg choice="req"> KEYNAME</arg>
```
Comment (jlebon) [inline man/ostree-config.xml]: Aside: I'm confused now why we use both the `GROUPNAME` and `SECTIONNAME` nouns when they're the same thing, right? I'd vote for harmonizing those to make it the same. Anyway, just writing this down so I don't forget, doesn't have to be in this patch!

File: tests/test-config.sh [hunk 2]
```diff
@@ -53,3 +53,20 @@ assert_file_has_content repo/config "Nightly Flathub"
 assert_file_has_content repo/config "false"
 assert_file_has_content repo/config "http://example.com/ostree/"
 echo "ok config set"
+
+${CMD_PREFIX} ostree config --repo=repo set core.lock-timeout-secs 60
+assert_file_has_content repo/config "lock-timeout-secs=60"
+${CMD_PREFIX} ostree config --repo=repo unset core.lock-timeout-secs
+assert_not_file_has_content repo/config "lock-timeout-secs=60"
```
Comment (jlebon) [inline tests/test-config.sh]: A stronger check here would be that the key doesn't exist at all (i.e. just `lock-timeout-secs=`). And maybe corroborate that with sanity checking that `ostree config get` erroring out?

**Comment by mwleeds** [comment]
I don't really have time to work on this right now but just interesting to note, `flatpak config` has a `--unset` option

File: man/ostree-config.xml (hunk 1 above)
Comment (mwleeds) [reply man/ostree-config.xml]: Added a commit to fix this

File: tests/test-config.sh (hunk 2 above)
Comment (mwleeds) [reply tests/test-config.sh]: You're right; I made both changes and force-pushed the branch

**Comment by mwleeds** [comment]
This is ready for re-review now, and has a new commit to check for too many arguments. Sorry for the delay

**Comment by jlebon** [comment]
This looks great to me! Will leave it open for a bit to give others time to review.

**Comment by cgwalters** [comment]
@rh-atomic-bot r+ 6b417c3

**Pull request closed without merging**